### PR TITLE
Static load API with taging

### PR DESCRIPTION
**Pull request by rizakii**
- IoTAPIstored with tag on nsuserdefault.

### CODE REVIEW

File: IoTCloudSDK/IoTCloudSDK/IoTCloudAPI.swift [hunk 1]
```diff
@@ -382,9 +393,31 @@ public class IoTCloudAPI: NSObject, NSCoding {
...
+    - Returns: IoTCloudAPI instance.
+    */
+    public static func loadWithStoredInstance(tag : String? = nil) throws -> IoTCloudAPI?{
+
+        var savedIoTAPI: IoTCloudAPI?
+        let key = IoTCloudAPI.getSharedPreferencesKey(tag)
+
+        // try to get iotAPI from NSUserDefaults
+        if let data = NSUserDefaults.standardUserDefaults().objectForKey(key) as? NSData {
+            savedIoTAPI = NSKeyedUnarchiver.unarchiveObjectWithData(data) as? IoTCloudAPI
```
Comment (yongpingchen) [inline IoTCloudSDK/IoTCloudSDK/IoTCloudAPI.swift]: I think  invalid error should be thrown if savedIoTAPI is not IoTCloudAPI

File: IoTCloudSDK/IoTCloudSDK/IoTCloudAPI.swift [hunk 2]
```diff
@@ -7,7 +7,15 @@ import Foundation
 
 /** Class provides API of the IoTCloud. */
 public class IoTCloudAPI: NSObject, NSCoding {
-    
+
+    private static let SHARED_PREFERENCES_KEY_INSTANCE = "IoTCloudAPI_INSTANCE"
```
Comment (yongpingchen) [inline IoTCloudSDK/IoTCloudSDK/IoTCloudAPI.swift]: shared preference is term of Android. Let us make it iOS like. What do you think? :)

Comment (rizakii) [inline IoTCloudSDK/IoTCloudSDK/IoTCloudAPI.swift]: @yongpingchen  how about `SHARED_NSUSERDEFAULT_KEY_INSTANCE`

File: IoTCloudSDK/IoTCloudSDK/IoTCloudAPI.swift [hunk 3]
```diff
@@ -382,9 +392,71 @@ public class IoTCloudAPI: NSObject, NSCoding {
...
         return newIotapi
     }
-    
+
+    /** Try to load the instance of IoTCloudAPI using stored serialized instance.
+
+    - Parameter tag: tag of the IoTCloudAPI instance
+    - Returns: IoTCloudAPI instance.
+    */
+    public static func loadWithStoredInstance(tag : String? = nil) throws -> IoTCloudAPI?{
```
Comment (yongpingchen) [inline IoTCloudSDK/IoTCloudSDK/IoTCloudAPI.swift]: why accept nil for tag? Since nil will throw error anyway.

**Comment by yongpingchen** [comment]
I think in the tests, we should not use the singleton NSUserDefaults.standardUserDefaults(). Like sharesession, we may need independence injection.

**Comment by yongpingchen** [comment]
Please add test for INVALID_STORED_API

**Comment by yongpingchen** [comment]
Looks fine to me

**Pull request merged**